### DISCUSSION TITLE
web page is displaying beyond the display size

### DISCUSSION THREAD

**Discussion by Ravikasyap** [Igalia/cog#542] [Q&A] [answered]
Hi, 
I am running yocto kirkstone branch with meta-webkit (Kirkstone brnach ) and added below line in local.conf file. 

CORE_IMAGE_EXTRA_INSTALL:append =" wpewebkit cog imx-gst1.0-plugin"                                                                     
PREFERRED_PROVIDER_virtual/wpebackend = "wpebackend-fdo"

After build the image running below command but the web page is beyond the screen size. I have set height and width of the screen but still the web page is beyond the display size. Kindly see the attachment.

root@imx8mm-var-dart:~# export COG_PLATFORM_FDO_VIEW_HEIGHT=480

root@imx8mm-var-dart:~# export COG_PLATFORM_FDO_VIEW_WIDTH=800

root@imx8mm-var-dart:~# cog google.com 

(cog:1697): GLib-GIO-WARNING **: 09:18:08.578: Your application does not implement g_application_activate() and has no handlers connected to the 'activate' signal.  It should do one of these.
Cog-Core-Message: 09:18:08.891: <http://google.com/> Load started.
Cog-Core-Message: 09:18:09.454: <http://www.google.com/> Redirected.
Cog-Core-Message: 09:18:09.456: <https://www.google.com/> Redirected.
Cog-Core-Message: 09:18:10.659: <https://www.google.com/> Loading...
Cog-Core-Message: 09:18:18.566: <https://www.google.com/> Loaded successfully.
![config detials](https://user-images.githubusercontent.com/99325959/210522681-2fd616e8-de8d-4ff8-8c22-e8cb3212708a.png)
![browser_after_setting](https://user-images.githubusercontent.com/99325959/210522686-f8f3cc7e-a792-4de3-8eb9-c8e3560c7c1b.jpg)

**Comment by philn** [comment, marked as answer]
```
export COG_PLATFORM_WL_VIEW_WIDTH=800
export COG_PLATFORM_WL_VIEW_HEIGHT=600
cog -P wl http://example.com
```

**Comment by aperezdc** [comment]
Alternatively, you can use `COG_PLATFORM_WL_VIEW_MAXIMIZE` to make Cog take all the space available without covering Weston's top bar:

```sh
COG_PLATFORM_WL_VIEW_MAXIMIZE=1 cog -P wl wpewebkit.org
```

For other available options, please check the [Wayland platform plug-in documentation](https://igalia.github.io/cog/platform-wl.html). I hope this helps.
